app/models/assignment.rb

```diff
@@ -752,11 +752,11 @@ def get_num_valid
 
   def get_num_marked(ta_id = nil)
     if ta_id.nil?
-      results_join = groupings.left_outer_joins(:current_result)
-      num_incomplete = results_join.where('results.id': nil)
-                                   .or(results_join.where('results.marking_state': 'incomplete'))
-                                   .count
-      get_num_assigned - num_incomplete
+      results_join = groupings.includes(:current_result).where('groupings.is_collected': true)
+      num_complete = results_join.where('results.id': nil)
+                                 .or(results_join.where('results.marking_state': 'complete'))
+                                 .count
+      num_complete
```

Comment:
You don't need a separate variable here; this entire block of code can be a single query (which you just return the value of)